src/tests/validity/function_optional_argument_array.ab

```diff
@@ -0,0 +1,13 @@
+// Output
+// 1100
+
+fun sum_array(a : [Num] = [Num]): Num {
```

Comment:
Why is the default value a type? Doesn't it raise any errors while compiling?

Reply:
I still dont understand this. Why do we allow for type to be a default value? What is the usecase?

Reply:
I don't know, I just migrated the test from the PR that was approved and were missing in the actual codebase.

Reply:
If this test succeeds, this is worth opening an issue about.

Reply:
tests right now works, if someone want to open a ticket about it to me is fine :-)